source/common/http/conn_manager_impl.cc

```diff
@@ -927,6 +939,12 @@ void ConnectionManagerImpl::ActiveStream::decodeData(ActiveStreamDecoderFilter*
       // a previous filter has added trailers.
       return;
     }
+
+    entry++;
+    if (entry != decoder_filters_.end() && (*entry)->stoppedAll()) {
```

Comment:
Similar comment here about how this happens. This pattern of looking at entry and checking if it's not stopped all happens multiple times. Would it make sense to wrap it in a function with a better name and more comments? That might make these cases easier to read.

Reply:
Comments added. Wrapped the stop-all state handling in a function. Thanks!